dropshot/examples/self-referential.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Oxide Computer Company
```

Comment:
Can you add a comment here describing this example?

Reply:
Done